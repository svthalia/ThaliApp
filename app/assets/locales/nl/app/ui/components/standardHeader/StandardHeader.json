{
  "Welcome": "Welkom",
  "Event": "Evenement",
  "Calendar": "Agenda",
  "Pizza": "Pizza",
  "Profile": "Profiel",
  "Registration": "Registratie",
  "Settings": "Instellingen",
  "Registrations": "Inschrijvingen",
  "Photos": "Foto's",
  "Album": "Album"
}
