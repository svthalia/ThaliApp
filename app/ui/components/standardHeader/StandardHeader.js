import React from 'react';
import { SafeAreaView, Text, View } from 'react-native';
import StatusBar from '@react-native-community/status-bar';
import { withTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

import { withNavigation } from 'react-navigation';
import Colors from '../../style/Colors';
import styles from './style/StandardHeader';
import IconButton from '../button/IconButton';

const sceneToTitle = (routeName, t) => {
  switch (routeName) {
    case 'Welcome':
      return t('Welcome');
    case 'Event':
      return t('Event');
    case 'Calendar':
      return t('Calendar');
    case 'Pizza':
      return t('Pizza');
    case 'Profile':
      return t('Profile');
    case 'Registration':
      return t('Registration');
    case 'Photos':
      return t('Photos');
    case 'PhotoAlbum':
      return t('Album');
    case 'Settings':
      return t('Settings');
    case 'EventAdmin':
      return t('Registrations');
    case 'PizzaAdmin':
      return t('Orders');
    default:
      return 'ThaliApp';
  }
};

const StandardHeader = props => (
  <View>
    <View style={styles.statusBar}>
      <StatusBar
        backgroundColor={Colors.semiTransparent}
        translucent
        animated
        barStyle="light-content"
      />
    </View>
    <SafeAreaView style={styles.safeArea}>
      <View style={styles.appBar}>
        <IconButton
          onPress={() => (props.menu ? props.navigation.toggleDrawer() : props.navigation.goBack())}
          name={props.menu ? 'menu' : 'arrow-back'}
          style={styles.iconButton}
          iconStyle={styles.icon}
        />
        <Text style={styles.title}>
          {sceneToTitle(props.navigation.state.routeName, props.t)}
        </Text>
        <View style={styles.rightView}>
          {props.rightView}
        </View>
      </View>
    </SafeAreaView>
  </View>
);

StandardHeader.propTypes = {
  menu: PropTypes.bool,
  rightView: PropTypes.element,
  t: PropTypes.func.isRequired,
  navigation: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

StandardHeader.defaultProps = {
  rightView: null,
  menu: false,
};

const StandardHeaderContainer = withNavigation(withTranslation('ui/components/standardHeader/StandardHeader')(StandardHeader));
export default StandardHeaderContainer;

export function withStandardHeader(Component, menu) {
  return props => (
    <View style={styles.rootWrapper}>
      <StandardHeaderContainer menu={menu} />
      <Component
        {...props}
      />
    </View>
  );
}
