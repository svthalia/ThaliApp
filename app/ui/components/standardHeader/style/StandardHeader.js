import { Platform } from 'react-native';
import StatusBar from '@react-native-community/status-bar';
import DeviceInfo from 'react-native-device-info';

import StyleSheet from '../../../style/StyleSheet';
import Colors from '../../../style/Colors';

// eslint-disable-next-line no-nested-ternary
export const STATUSBAR_HEIGHT = Platform.OS === 'ios'
  ? DeviceInfo.hasNotch() ? 44 : 20 : StatusBar.currentHeight;
export const APPBAR_HEIGHT = Platform.OS === 'ios' ? 44 : 56;

export const TOTAL_BAR_HEIGHT = APPBAR_HEIGHT + STATUSBAR_HEIGHT;

const styles = StyleSheet.create({
  rootWrapper: {
    flex: 1,
  },
  statusBar: {
    height: STATUSBAR_HEIGHT,
    backgroundColor: Colors.magenta,
    android: {
      elevation: 4,
    },
  },
  safeArea: {
    backgroundColor: Colors.magenta,
    android: {
      height: APPBAR_HEIGHT,
      elevation: 4,
    },
    ios: {
      borderStyle: 'solid',
      borderBottomColor: Colors.darkMagenta,
      borderBottomWidth: 1,
    },
  },
  appBar: {
    backgroundColor: Colors.magenta,
    height: APPBAR_HEIGHT,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  title: {
    color: Colors.white,
    android: {
      fontSize: 20,
      fontFamily: 'sans-serif-medium',
    },
    ios: {
      position: 'absolute',
      left: 40,
      right: 40,
      zIndex: 0,
      textAlign: 'center',
      fontSize: 18,
      fontFamily: 'System',
      fontWeight: '600',
    },
  },
  iconButton: {
    zIndex: 2,
  },
  icon: {
    android: {
      paddingLeft: 20,
      paddingRight: 32,
    },
    ios: {
      paddingLeft: 10,
      paddingRight: 16,
    },
    color: Colors.white,
  },
  rightView: {
    position: 'absolute',
    right: 0,
  },
});

export default styles;
