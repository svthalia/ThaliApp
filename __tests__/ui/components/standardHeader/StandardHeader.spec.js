import React from 'react';
import { Platform } from 'react-native';
import renderer from 'react-test-renderer';
import StandardHeader from '../../../../app/ui/components/standardHeader/StandardHeader';

jest.mock('react-navigation', () => ({
  withNavigation: (component) => component,
}));

const mockNavigation = {
  state: 'mock',
};

jest.mock('../../../../app/ui/components/button/IconButton', () => 'IconButton');

jest.mock('@react-native-community/status-bar', () => 'StatusBar');

describe('StandardHeader component', () => {
  const initialState = {
    navigation: {
      currentScene: 'home',
      previousScenes: [],
      loggedIn: false,
      drawerOpen: false,
    },
    session: {
      displayName: 'Lorem ipsum',
      photo: 'photo',
      token: 'token123',
    },
  };

  it('renders correctly on iOS', () => {
    Platform.OS = 'ios';
    const tree = renderer
      .create(<StandardHeader store={initialState} navigation={mockNavigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly on Android', () => {
    Platform.OS = 'android';
    const tree = renderer
      .create(<StandardHeader store={initialState} navigation={mockNavigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
