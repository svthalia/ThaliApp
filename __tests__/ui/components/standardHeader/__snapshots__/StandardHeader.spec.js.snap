// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardHeader component renders correctly on Android 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#E62272",
        "height": 20,
      }
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#E62272",
        "borderBottomColor": "#C2185B",
        "borderBottomWidth": 1,
        "borderStyle": "solid",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E62272",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "height": 44,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "zIndex": 2,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 24,
              },
              Object {
                "color": "#FFFFFF",
                "paddingLeft": 10,
                "paddingRight": 16,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 18,
            "fontWeight": "600",
            "left": 40,
            "position": "absolute",
            "right": 40,
            "textAlign": "center",
            "zIndex": 0,
          }
        }
      >
        ThaliApp
      </Text>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 0,
          }
        }
      />
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`StandardHeader component renders correctly on iOS 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#E62272",
        "height": 20,
      }
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#E62272",
        "borderBottomColor": "#C2185B",
        "borderBottomWidth": 1,
        "borderStyle": "solid",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E62272",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "height": 44,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "zIndex": 2,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 24,
              },
              Object {
                "color": "#FFFFFF",
                "paddingLeft": 10,
                "paddingRight": 16,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 18,
            "fontWeight": "600",
            "left": 40,
            "position": "absolute",
            "right": 40,
            "textAlign": "center",
            "zIndex": 0,
          }
        }
      >
        ThaliApp
      </Text>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 0,
          }
        }
      />
    </View>
  </RCTSafeAreaView>
</View>
`;
